api/v1alpha1/vineyardruntime_types.go

```diff
@@ -0,0 +1,257 @@
+	// Support the following options.
+	//
+	//   etcd.prefix: (String) the prefix of etcd key for vineyard objects
+	//
+	// Default value is as follows.
+	//
+	//   etcd.prefix: "/vineyard"
+	//
+	// +optional
+	Options map[string]string `json:"options,omitempty"`
+}
+
+// MasterSpec defines the configurations for Vineyard Master component
+// which is also regarded as the Etcd component in Vineyard.
+// For more info about Vineyard, refer to <a href="https://v6d.io/">Vineyard</a>
+type MasterSpec struct {
+	// The component configurations for Vineyard Master
+	// +optional
+	VineyardCompTemplateSpec `json:",inline"`
+
+	// ExternalEndpoint defines the configurations for external etcd cluster
+	// Default is not set
+	// If set, the Vineyard Master component will not be deployed,
+	// which means the Vineyard Worker component will use an external Etcd cluster.
+	// E,g.
+	//   endpoint:
+	//     uri: "etcd-svc.etcd-namespace.svc.cluster.local:2379"
+	//     encryptOptions:
+	//       - name: access-key
+	// 		   valueFrom:
+	//           secretKeyRef:
+	//             name: etcd-secret
+	//			   key: accesskey
+	// +optional
+	ExternalEndpoint ExternalEndpointSpec `json:"endpoint,omitempty"`
+}
+
+// VineyardFuseSpec holds the configurations for vineyard fuse
+type VineyardFuseSpec struct {
+	// Image for Vineyard Fuse
+	// Default is `vineyardcloudnative/vineyard-mount-socket`
+	// +optional
+	Image string `json:"image,omitempty"`
+
+	// Image Tag for Vineyard Fuse
+	// Default is `latest`
+	// +optional
+	ImageTag string `json:"imageTag,omitempty"`
+
+	// Image pull policy for Vineyard Fuse
+	// Default is `IfNotPresent`
+	// Available values are `Always`, `IfNotPresent`, `Never`
+	// +optional
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	// CleanPolicy decides when to clean Vineyard Fuse pods.
+	// Currently Fluid supports two policies: OnDemand and OnRuntimeDeleted
+	// OnDemand cleans fuse pod once th fuse pod on some node is not needed
+	// OnRuntimeDeleted cleans fuse pod only when the cache runtime is deleted
+	// Defaults to OnRuntimeDeleted
+	// +optional
+	CleanPolicy FuseCleanPolicy `json:"cleanPolicy,omitempty"`
+
+	// Resources contains the resource requirements and limits for the Vineyard Fuse.
+	// Default is not set.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+// VineyardRuntimeSpec defines the desired state of VineyardRuntime
+type VineyardRuntimeSpec struct {
+	// Master holds the configurations for Vineyard Master component
+	// Represents the Etcd component in Vineyard
+	// +optional
+	Master MasterSpec `json:"master,omitempty"`
+
+	// Worker holds the configurations for Vineyard Worker component
+	// Represents the Vineyardd component in Vineyard
+	// +optional
+	Worker VineyardCompTemplateSpec `json:"worker,omitempty"`
+
+	// Fuse holds the configurations for Vineyard Fuse
+	// +optional
+	Fuse VineyardFuseSpec `json:"fuse,omitempty"`
```

Comment:
@dashanji Would it be better to use something like `VineyardSockSpec` or `VineyardClientSpec` instead of `VineyardFuseSpec`? From my understanding, we are not going to use the real vineyard fuse but using the vineyard socket to mimic a fake fuse.

So what I'm thinking is we can keep the `VineyardRuntime.spec.fuse` for API consistency with other runtimes. But in Fluid's codebase, we can avoid using `Fuse` as the golang struct's name. For example, something like:
```
// Fuse holds the configurations for Vineyard client socket.
// Note that  the "Fuse" here is kept just for API consistency, VineyardRuntime mount a socket file instead of a FUSE filesystem to make data cache available.
// <Maybe some Vineyard arch document here>
Fuse VineyardSockSpec `json:"fuse,omitempty"`
```

Reply:
Make sense to me. Thanks @TrafalgarZZZ

Reply:
@dashanji Sorry for responding late. Thanks! The PR LGTM.